x-pack/legacy/plugins/snapshot_restore/server/lib/get_managed_repository_name.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const getManagedRepositoryName = async (
+  callWithInternalUser: any
+): Promise<string | undefined> => {
+  try {
+    const { persistent, transient, defaults } = await callWithInternalUser('cluster.getSettings', {
+      filterPath: '*.*managed_repository',
+      flatSettings: true,
+      includeDefaults: true,
+    });
+    const { 'cluster.metadata.managed_repository': managedRepositoryName = undefined } = {
+      ...defaults,
+      ...persistent,
+      ...transient,
+    };
+    return managedRepositoryName;
+  } catch (e) {
+    // Silently swallow error and return undefined
```

Comment:
Not a blocker but I'm still not quite getting this. Can you help me understand over Zoom?

Reply:
if the request to get cluster settings fails for some reason, simply return undefined for `managedRepositoryName`. downstream logic will act as if there is no `managed_repository` setting, which is the behavior for a non-cloud instance such as local dev environment

Reply:
> for some reason

I think that's what sets my spidey sense tingling. If an error is an expected part of the interface of something we're consuming inside of the `try`, then I can see swallowing it because it has some kind of meaning that we've identified as inconsequential. But if the error is unexpected then that's something I'd expect to bubble up and cause a 500. Otherwise if users are on Cloud and we get an error here, won't they be able to delete snapshots?

Reply:
I view this as an extreme edge case because
1. this is call made with `callWithInternalUser` (superuser), so it won't fail due to things like permissions
2. the call is made to  the cluster API `/_cluster/settings`, if this call fails with a superuser, I expect the user's instance is having larger problems than this UI

I don't expect the `catch` logic to really ever be hit for the above reasons. previously this method was not wrapped in `try...catch`, and any errors would prevent the downstream APIs from returning _any_ information.

> if users are on Cloud and we get an error here, won't they be able to delete snapshots?

the minimal possibility of getting an error here notwithstanding, Cloud users are able to override any UI blocks we put into place if they want to 🙂

Reply:
> Cloud users are able to override any UI blocks we put into place if they want to

You're right, I should have phrased my point better: won't the UI no longer distinguish between Cloud and non-Cloud repos/snaps? So won't the user be in a much more dangerous situation?

> any errors would prevent the downstream APIs from returning any information.

I see. Do I understand the reason correctly: we've identified the information here as optional, and if this part of the handler fails then the client can still safely continue without the information this request provides. If this is the reasoning, then can we change the comment to communicate that intention, so the comment describes the "why" instead of the "what"?